porcupine/utils.py

```diff
@@ -144,9 +145,13 @@ def quote(string: str) -> str:
 
 
 else:
-    from shlex import quote
+    quote = shlex.quote
 
-    quote = quote  # silence pyflakes warning
+
+# TODO: document this
+def format_command(command: str, substitutions: dict[str, Any]) -> list[str]:
+    parts = shlex.split(command, posix=(sys.platform != "win32"))
+    return [part.format_map(substitutions) for part in parts]
```

Comment:
Would be nice to add tests for this, now that it's separate from the rest of the code.